ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/HARouting.java

```diff
@@ -273,29 +276,37 @@ static void executeOrRouteQuery(
         pullQueryMetrics
             .ifPresent(queryExecutorMetrics -> queryExecutorMetrics.recordLocalRequests(1));
         pullPhysicalPlan.execute(locations, pullQueryQueue,  rowFactory);
+      } catch (StandbyFallbackException e) {
+        LOG.error("Standby Fallback: Error executing query {} locally at node {} with exception",
+            statement.getStatementText(), node, e.getCause());
+        throw e;
       } catch (Exception e) {
-        LOG.error("Error executing query {} locally at node {} with exception",
+        LOG.error("Error executing query {} locally at node {}",
             statement.getStatementText(), node, e.getCause());
         throw new KsqlException(
-            String.format("Error executing query %s locally at node %s",
-                          statement.getStatementText(), node),
+            String.format("Error executing query locally at node %s: %s", node.location(),
+                e.getMessage()),
             e
         );
       }
     } else {
       try {
         LOG.debug("Query {} routed to host {} at timestamp {}.",
-                  statement.getStatementText(), node.location(), System.currentTimeMillis());
+            statement.getStatementText(), node.location(), System.currentTimeMillis());
         pullQueryMetrics
             .ifPresent(queryExecutorMetrics -> queryExecutorMetrics.recordRemoteRequests(1));
         forwardTo(node, locations, statement, serviceContext, pullQueryQueue, rowFactory,
             outputSchema);
+      } catch (StandbyFallbackException e) {
+        LOG.error("Standby Fallback: Error forwarding query {} to node {} with exception {}",
+            statement.getStatementText(), node, e.getCause());
+        throw e;
```

Comment:
Since on the caller we are going to swallow the `StandbyFallbackException` and retry on standbys, how about letting `executeOrRouteQuery` to return a result (for now, just boolean is okay; in general, maybe a enum can do more things) where the caller can check on the `futures`, instead of capturing and opening the exception envelope to check?

Reply:
Sure, good suggestion.  I just created an enum to make it a little easier to read.